MovieTime/MovieTime/Service/MTMovieDBService.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  MTMovieDBService.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 08/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import Foundation
+
+protocol MTMovieDBServiceProtocol: class {
+    init(service: MTServiceProtocol, helper: MTServiceHelperProtocol)
+
+    func requestConfiguration(completion: ((Data?, MTNetworkError?) -> Void)?)
+    func requestGenresList(completion: ((Data?, MTNetworkError?) -> Void)?)
+    func requestUpcomingMovies(forPage page: Int, completion: ((Data?, MTNetworkError?) -> Void)?)
+    func requestMovieDetail(forId movieId: Int, completion: ((Data?, MTNetworkError?) -> Void)?)
+}
+
+final class MTMovieDBService: MTMovieDBServiceProtocol {
+
+    private let urlServer = URL(string: "https://api.themoviedb.org/3")
+    private let currentService: MTServiceProtocol
+    private let serviceHelper: MTServiceHelperProtocol
+
+    private let apiKey = "1f54bd990f1cdfb230adb312546d765d"
+    private let language = "en-US"
```

Comment:
in a future version of this app, what is the architecture you envision to add support to other languages?

Reply:
The language should be pass throught the service initializer.
Now, to get the correct value I imagine 2 ways:
1 - Get the user preference in the Settings screen;
2 - Or use the value from device.

With this value, when creating the service it is possible to use it in the requests, similar to the `MTMovieDBImageService` initializer.